app/controllers/drinks_controller.rb

```diff
@@ -22,6 +21,38 @@ def show
     @drink = Drink.find(params[:id])
   end
 
+  def edit
+    @drink = Drink.find(params[:id])
+    @liquors = Liquor.all
+    if !signed_in?
+      flash[:error] = "You must be signed in to do that"
+      redirect_to drink_path(@drink)
+    elsif @drink.user != current_user
+      flash[:error] = "You cannot edit someone else's drink"
+      redirect_to drink_path(@drink)
```

Comment:
Is a `redirect_to` command (and a new HTTP request) the intended action here? Would it be better to keep the user in their existing client state with a `render`?